lib/src/value/calculation.dart

```diff
@@ -223,6 +605,83 @@ class SassCalculation extends Value {
   /// simplification.
   SassCalculation._(this.name, this.arguments);
 
+  // Returns the value coerced to numbers units.
+  static SassNumber _matchUnits(double value, SassNumber number) {
+    return SassNumber.withUnits(value, numeratorUnits: number.numeratorUnits);
+  }
+
+  // Returns whether numbers are unsupported units.
+  static bool _arePossiblyCompatible(SassNumber first, Object? second) {
+    try {
+      if (second is SassNumber) second.coerceToMatch(first);
+      return false;
+    } on SassScriptException {
+      return true;
+    }
+  }
+
+  /// Returns a rounded [number] based on a selected rounding [strategy],
+  /// to the nearest integer multiple of [step].
+  static SassNumber _roundWithStep(
+      SassString strategy, SassNumber number, SassNumber step) {
+    if (!{'nearest', 'up', 'down', 'to-zero'}.contains(strategy.text)) {
+      throw ArgumentError(
+          "$strategy must be either nearest, up, down or to-zero.");
+    }
+
+    if ((number.value.isInfinite && step.value.isInfinite) ||
+        ((step.value == 0 || number.value.isNaN) || step.value.isNaN)) {
+      return _matchUnits(double.nan, number);
+    }
+    if (number.value.isInfinite) return number;
+
+    if (step.value.isInfinite) {
+      switch (strategy.text) {
```

Comment:
You can make this a switch expression, since it's exhaustive and every branch returns a value.